src/mqttgw_ratelimitstate.erl

```diff
@@ -0,0 +1,90 @@
+-module(mqttgw_ratelimitstate).
+
+%% API
+-export([
+    new/0,
+    get_dirty/1,
+    get/2,
+    put_dirty/2,
```

Comment:
why export get_dirty and put_dirty?

Reply:
Their purpose is not only internal use. They do not modify an internal state and useful for inspection purposes.